zeus/optimizer/batch_size/server/router.py

```diff
@@ -22,6 +26,10 @@
 from zeus.util.logging import get_logger
 
 app = FastAPI()
+# Global variable across different requests: https://github.com/tiangolo/fastapi/issues/592
+# We lock the job before we make any modification to prevent any concurrent bugs.
+app.job_locks = defaultdict(asyncio.Lock)
```

Comment:
It's time to implement `delete_job`, which will remove that entry from `job_locks`. I thought of this because as the server runs and more and more jobs run with it, the size of this dictionary will just keep growing.